### PR TITLE
kotlin app bindings

### PR DESCRIPTION
**Pull request by krisbitney**

### CODE REVIEW

**Comment by dOrgJelli** [comment]
In order to show the e2e developer experience, can we please create a template project that uses these bindings?

Similar to: https://github.com/polywrap/cli/blob/origin-dev/packages/templates/app/typescript/src/index.ts

**Comment by krisbitney** [comment]
> In order to show the e2e developer experience, can we please create a template project that uses these bindings?
> 
> Similar to: https://github.com/polywrap/cli/blob/origin-dev/packages/templates/app/typescript/src/index.ts

Template PR is here: https://github.com/polywrap/cli/pull/1867

File: implementations/app-kotlin/src/__tests__/cases/000-sanity/output/types.kt [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+) {
+    companion object {
+        val uri: Uri = Uri("testimport.uri.eth")
+    }
+
+    protected val defaultClient: Invoker by lazy {
+        polywrapClient { addDefaults() }
+    }
+    protected val defaultUri: Uri? = null
+    protected val defaultEnv: TestImportEnv? = null
```
Comment (Niraj-Kamdar) [inline implementations/app-kotlin/src/__tests__/cases/000-sanity/output/types.kt:199]: As a non Kotlin dev, I am just curious why we aren't using same pattern as other clients where we use constructor and abstract methods that can be extended

**Review by dOrgJelli — APPROVED** [review]
This LGTM, we can continue to make improvements to it post-merge via new versions on wrapscan.

File: implementations/app-kotlin/src/__tests__/cases/000-sanity/output/types.kt (hunk 1 above)
Comment (krisbitney) [reply implementations/app-kotlin/src/__tests__/cases/000-sanity/output/types.kt:199]: In Kotlin, abstract members can only be in an abstract class. If a user wants to, they can extend the class and override the members `defaultClient`, `defaultUri`, and `defaultEnv`

**Pull request merged**